packages/timmy/timmy.1.0.0/opam

```diff
@@ -0,0 +1,50 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Time and calendar library"
+maintainer: ["mefyl <mefyl@gruntech.org>"]
+authors: ["mefyl <mefyl@gruntech.org>"]
+license: "LGPL-3.0-or-later"
+homepage: "https://github.com/mefyl/timmy"
+bug-reports: "https://github.com/mefyl/timmy/issues"
+depends: [
+  "dune" {>= "3.0"}
+  "base"
+  "fmt"
+  "ocaml" {>= "4.08.0"}
+  "ppx_here"
+  "ppx_deriving"
+  "ptime"
+  "alcotest" {with-test}
+  "odoc" {with-doc}
+  "cmdliner" {dev}
+  "opam-file-format" {dev}
+  "sexplib" {dev}
+  "stdio" {dev}
+  "shexp" {dev}
+]
+depopts: [
+  "js_of_ocaml"
+  "schematic" {>= "0.17.0" & < "0.18.0"}
+]
```

Comment:
In that case this is used to add instrumentation to the code when schematic (unreleased for now) is available, my idea there is that if an uncompatible version is installed, we might as well still allow installation and not add the instrumentation ? Granted it can make debugging why the instrumentation is or isn't present a bit harder, but sounds more flexible wrt the set of installable packages.

We don't really care internally as we always use the latest versions, so I'm leaving the final decision to you.

Reply:
The problem is that looking at https://github.com/mefyl/timmy/blob/7120fc0462f8d30e171e761f7ee7110b9d8c394f/timmy/lib/dune#L13 compilation will fail if schematic is detected but the API isn't compatible. Also with the way the depopt is currently used the package might be installed in parallel with `timmy` and cause problem there as well

Reply:
I think dune does not `select` the file if the optional dependency doesn't match the constraint ?

In any case I'll trust your judgment regarding what will make the package more usable, modifications applied.

Reply:
> I think dune does not `select` the file if the optional dependency doesn't match the constraint ?

What do you mean by that? Dune doesn't know anything about the constraints from the opam file (even when defined in the dune-project file for autogeneration purpose, if that's what you were thinking of)

Reply:
I thought it did, I suppose it *could* thanks to the `dune-project`. In any case if it does not I stand corrected!